plugins/woocommerce-blocks/assets/js/blocks/product-template/utils.tsx

```diff
@@ -366,3 +366,32 @@ export const useProductCollectionQueryContext = ( {
 		return queryContext;
 	}, [ queryContextIncludes, productCollectionBlockAttributes ] );
 };
+
+export const parseTemplateSlug = ( rawTemplateSlug = '' ) => {
+	const categoryPrefix = 'category-';
+	const productCategoryPrefix = 'taxonomy-product_cat-';
+	const productTagPrefix = 'taxonomy-product_tag-';
+
+	if ( rawTemplateSlug.startsWith( categoryPrefix ) ) {
+		return {
+			taxonomy: 'category',
+			slug: rawTemplateSlug.replace( categoryPrefix, '' ),
+		};
+	}
```

Comment:
As I mentioned in another comment, if we decide not to handle WordPress default categories, we can remove this code 🙂